en/nodes-hw-scripts.md

```diff
@@ -0,0 +1,124 @@
+Title: Hardware Testing Scripts
+table_of_contents: True
+
+# Hardware Testing Scripts
+
+[Hardware Testing][hardware-testing] scripts can be run directly after [node
+commissioning][commission-nodes], or from a node in a ready, deployed or broken
+state. Your own scripts can be uploaded and run during commissioning or the
+testing phase, alongside scripts [bundled with MAAS][bundled-scripts].
+
+Scripts typically evaluate system hardware, collate data and report results
+back to MAAS. As scripts are run within an ephemeral Ubuntu environment, they
+can be written in Bash, Python, Perl, or any other language installed via a
+*deb* or *Snap* package.
+
+## Script example 
+
+As a simple example, here's a functional Bash script replicating part of the
+**stress-ng** script bundled with MAAS:
+
+```bash
+#!/bin/bash -e
+# --- Start MAAS 1.0 script metadata ---
+# name: stress-ng-cpu-test
+# title: CPU validation
+# description: Run stress-ng memory tests for 5 minutes.
+# script_type: test
+# hardware_type: cpu
+# tags: cpu
+# timeout: 00:05:00
+# --- End MAAS 1.0 script metadata ---
+
+sudo apt-get --assume-yes install stress-ng
+sudo -n stress-ng --matrix 0 --ignite-cpu --log-brief --metrics-brief --times \
+    --tz --verify --timeout 2m
+```
+
+The above Bash script contains comment-delineated metadata plus two lines of
+functionality, the first to install the `stress-ng` package (a CPU stress-test
+utility) and the second to execute **stress-ng** with various arguments.
+
+The following metadata can be included:
+
+- `name`: Corresponds to the script's filename
+- `title`: Human-friendly descriptive version of name, used within the web UI
+- `description`: Brief outline of what the script does
+- `tags`:  List of tags associated with the script.
+- `script-type`,`hardware_type`: Broad categories for the kind of script and
+  what the script tests
```

Comment:
Re these predefined anywhere? Are there guidelines? I'm guessing not...

Reply:
no, unfortunately. I've added a TODO as there are a few things like this in these docs.